File: leetcode/python/7.reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        rev = 0
        y = abs(x)

        while y:
            rev = rev*10 + y % 10
            y //= 10

        sign = -1 if x < 0 else 1

        if(rev > 0x7fffffff):
            return 0

        return sign * rev

File: leetcode/cpp/7.reverse-integer.cpp
class Solution {
public:
	int reverse(int _x) {
		bool neg = _x < 0 ? true : false;
        
        
		int64_t x = abs<int64_t>(_x);
		int res = 0;
		while (x)
		{
			int t = x % 10;
			x /= 10;
            if((INT_MAX - t) / 10 < res){
                return 0;
            }
			res = res * 10 + t;
		}
		res = (neg ? -res : res);
		return res;
	}
};
